projects/ui/src/Apis/gg_hooks.ts

```diff
@@ -506,3 +507,142 @@ export function useDeleteOAuthMutation(appId: string) {
+
+export type CreateUpdateAppMetadataParams = MutationWithArgs<{
+  appId: string;
+  rateLimit?: RateLimit;
+  customMetadata?: Record<string, string>;
+}>;
+
+export function useCreateAppMetadataMutation() {
+  const { latestAccessToken } = useContext(AuthContext);
+  const { mutate } = useSWRConfig();
+  const createAppMetadata = async (
+    _: string,
+    { arg }: CreateUpdateAppMetadataParams
+  ) => {
+    const req: Record<string, any> = { appId: arg.appId };
+    if (arg.customMetadata !== undefined) {
+      req.customMetadata = arg.customMetadata;
+    }
+    if (arg.rateLimit !== undefined) {
+      req.rateLimit = arg.rateLimit;
+    }
+    await fetchJSON(`/apps/${arg.appId}/metadata`, {
+      method: "POST",
+      headers: getLatestAuthHeaders(latestAccessToken),
+      body: JSON.stringify(req),
+    });
+    mutate(TEAM_APPS_SWR_KEY);
+    mutate(`/apps/${arg.appId}`);
+  };
+  return useSWRMutation(`create-app-metadata`, createAppMetadata);
+}
+
+// -------------------------------- //
+// region (Admin) Update App Metadata
+
+export function useUpdateAppMetadataMutation() {
+  const { latestAccessToken } = useContext(AuthContext);
+  const { mutate } = useSWRConfig();
+  const fetcher = async (_: string, { arg }: CreateUpdateAppMetadataParams) => {
+    const req: Record<string, any> = { appId: arg.appId };
+    if (arg.customMetadata !== undefined) {
+      req.customMetadata = arg.customMetadata;
+    }
+    if (arg.rateLimit !== undefined) {
+      req.rateLimit = arg.rateLimit;
+    }
+    await fetchJSON(`/apps/${arg.appId}/metadata`, {
+      method: "PUT",
+      headers: getLatestAuthHeaders(latestAccessToken),
+      body: JSON.stringify(req),
+    });
+    mutate(TEAM_APPS_SWR_KEY);
+    mutate(`/apps/${arg.appId}`);
+  };
+  return useSWRMutation("update-app-metadata", fetcher);
+}
+
+// -------------------------------- //
+// region (Admin) Create Subscription Metadata
+
+export type CreateUpdateSubscriptionMetadataParams = MutationWithArgs<{
+  subscription: Subscription;
+  customMetadata?: Record<string, string>;
+  rateLimit?: RateLimit;
+}>;
+
+export function useCreateSubscriptionMetadataMutation() {
+  const { latestAccessToken } = useContext(AuthContext);
+  const { mutate } = useSWRConfig();
+  const fetcher = async (
+    _: string,
+    { arg }: CreateUpdateSubscriptionMetadataParams
+  ) => {
+    const req: Record<string, any> = { subscriptionId: arg.subscription.id };
+    if (arg.customMetadata !== undefined) {
+      req.customMetadata = arg.customMetadata;
+    }
+    if (arg.rateLimit !== undefined) {
+      req.rateLimit = arg.rateLimit;
+    }
+    await fetchJSON(`/subscriptions/${arg.subscription.id}/metadata`, {
+      method: "POST",
+      headers: getLatestAuthHeaders(latestAccessToken),
+      body: JSON.stringify(arg),
+    });
+    mutate(`/apps/${arg.subscription.applicationId}/subscriptions`);
+    mutate(`/subscriptions?status=${SubscriptionStatus.APPROVED}`);
```

Comment:
This feels weird, doing all 3 mutates at once.  Not a real issue, but could use a comment to explain such a strong action